packages/server/api/webscrape.py

```diff
@@ -1,20 +1,40 @@
 from http.server import BaseHTTPRequestHandler
+import urllib.request as request
+import re
 from urllib.parse import parse_qs
+from bs4 import BeautifulSoup
 
 class handler(BaseHTTPRequestHandler):
-
+  whitelisted_url_regexes = [
+    re.compile("^https:\/\/www.bonappetit.com")
+  ]
   def do_GET(self):
     queryString = self.path.split('webscrape?')
     if(len(queryString) > 1 and 'url=' in queryString[1]):
-        query = parse_qs(queryString[1])
-        self.send_response(200)
+      query = parse_qs(queryString[1])
+      url = query['url'][0]
+
+      if(not any(regex.match(url) for regex in self.whitelisted_url_regexes)):
+        self.send_response(500)
         self.send_header('Content-type', 'text/plain')
         self.end_headers()
-        self.wfile.write(str(query['url'][0]).encode())
+        self.wfile.write('Unsupported recipe site'.encode())
+        return
+
+      fp = request.urlopen(url)
+      mybytes = fp.read()
+      html_doc = mybytes.decode("utf8")
+      fp.close()
+
+      soup = BeautifulSoup(html_doc, 'html.parser')
```

Comment:
What is soup?

Reply:
soup's a library thats used for webscraping: https://www.crummy.com/software/BeautifulSoup/bs4/doc/

Reply:
I mean I figured, but more like what does this variable mean? Ahaha it might be better to have this named differently

Reply:
Changed to `html_parsed`